tests/lax_control_flow_test.py

```diff
@@ -538,6 +538,14 @@ def f(upper):
     with self.assertRaisesRegex(ValueError, "Can only use `unroll`"):
       f(10)
 
+  def test_fori_loop_unroll_true_returns_init_with_nonpositive_length(self):
```

Comment:
Suggestion: can we parameterize this test to negative lengths in all possible scenarios? rollend and unrolled, jit enabled and disabled?

Reply:
Caught a couple more cases when I added more test coverage. Namely, `scan` doesn't handle `length < 0`:

https://github.com/google/jax/blob/a6f26306b387ce082358e8e94fdcfd777773475f/jax/_src/lax/control_flow/loops.py#L1027

In that case, I return early, too.